### PR TITLE
Add fuzzing for BlockLoader endpoints

### PR DESCRIPTION
**Pull request by luckychess**
### Description of the Change

Fuzzing for block loader endpoints (**retrieveBlock** and **retrieveBlocks**).

### Benefits
Another endpoint is covered by fuzzing.

### Possible Drawbacks 
Hard to support until fuzzing in CI is not ready.

### Usage Examples or Tests 
```
cmake -DCMAKE_C_COMPILER=/usr/local/opt/llvm/bin/clang -DCMAKE_CXX_COMPILER=/usr/local/opt/llvm/bin/clang++  -DFUZZING=ON ..
make retrieve_block_fuzz
make retrieve_blocks_fuzz
```

### CODE REVIEW

File: test/fuzzing/retrieve_blocks_fuzz.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  static fuzzing::BlockLoaderFixture fixture;
+
+  if (size < 1) {
+    return 0;
+  }
+
+  iroha::network::proto::BlocksRequest request;
+  if (protobuf_mutator::libfuzzer::LoadProtoInput(true, data, size, &request)) {
+    grpc::ServerContext context;
+    MockServerWriter writer;
```
Comment (l4l) [inline test/fuzzing/retrieve_blocks_fuzz.cpp]: ServerWriter cannot be mocked :( I've tried with ServerWriterInterface, but it cannot be casted safely (e.g only via `reinterpret_cast`. Here's an example of such mock (path _test/module/vendor/grpc/grpc_mocks.hpp_):

```cpp
class MockServerWriter
    : public grpc::ServerWriterInterface<iroha::protocol::ToriiResponse> {
  MOCK_METHOD1(Write, void(iroha::protocol::ToriiResponse));
  MOCK_METHOD2(Write,
               bool(const iroha::protocol::ToriiResponse &,
                    grpc::WriteOptions));
  MOCK_METHOD0(SendInitialMetadata, void());
  MOCK_METHOD1(NextMessageSize, bool(uint32_t *));
};
```

But it's not really a good idea to use it. Need to think about that issue

File: test/fuzzing/retrieve_blocks_fuzz.cpp [hunk 2]
```diff
@@ -5,6 +5,17 @@
 
 #include "block_loader_fixture.hpp"
 
+namespace fuzzing {
+  class MockServerWriter
```
Comment (l4l) [inline test/fuzzing/retrieve_blocks_fuzz.cpp]: plz move it to `test/module/vendor/grpc_mocks.hpp`, I will also need that mock

File: test/fuzzing/retrieve_blocks_fuzz.cpp [hunk 3]
```diff
@@ -15,7 +26,12 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, std::size_t size) {
   iroha::network::proto::BlocksRequest request;
   if (protobuf_mutator::libfuzzer::LoadProtoInput(true, data, size, &request)) {
     grpc::ServerContext context;
-    fixture.block_loader_service_->retrieveBlocks(&context, &request, nullptr);
+    fuzzing::MockServerWriter serverWriter;
```
Comment (l4l) [inline test/fuzzing/retrieve_blocks_fuzz.cpp]: Should be static, I guess
Moreover some calls probably should be mocked (particularly the ones that return boolean)

Comment (luckychess) [reply test/fuzzing/retrieve_blocks_fuzz.cpp]: I don't think it should be static - we pass a non-const raw point to it and have no idea what will happen with it inside.
I'm not sure what call do you mean - the only one which returns boolean is ```Write``` with 2 args but we are not interested in it.

**Pull request merged**